### PR TITLE
[CDF-19684] Add Transformation Worker concurrencyPolicy

### PR DESCRIPTION
**Pull request by VerstraeteBert**
## Description
Adds concurrencyPolicy for Transformations:
```
    Determines the behavior of the task if the Transformation is already running.

    - FAIL: The task fails if another instance of the Transformation is currently running.
    - WAIT_FOR_CURRENT: The task will pause and wait for the already running Transformation to complete. Once completed, the task is completed. This mode is useful for preventing redundant Transformation runs.
    - RESTART_AFTER_CURRENT: The task waits for the ongoing Transformation to finish. After completion, the task restarts the Transformation. This mode ensures that the most recent data can be used by following tasks.
```

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version updated

### CODE REVIEW

File: cognite/client/_version.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
 from __future__ import annotations
 
-__version__ = "6.37.0"
+__version__ = "6.37.1"
```
Comment (doctrino) [inline cognite/client/_version.py:3]: Worthy of a minor bump :) We typically do that when we add a new property.

**Review by doctrino — COMMENTED** [review]
Just make it a major bump

**Comment by haakonvt** [comment]
> Just make it a major bump

**Minor** would be preferred 😅 (as his other comment states 😄 )

File: cognite/client/data_classes/workflows.py [hunk 2]
```diff
@@ -193,29 +194,52 @@ def dump(self, camel_case: bool = False) -> dict[str, Any]:
...
     def _load(cls, resource: dict | str, cognite_client: CogniteClient | None = None) -> TransformationTaskParameters:
         resource = json.loads(resource) if isinstance(resource, str) else resource
         return cls(
             resource["transformation"]["externalId"],
+            TransformationConcurrencyPolicy(resource["transformation"]["concurrencyPolicy"]),
         )
 
     def dump(self, camel_case: bool = False) -> dict[str, Any]:
-        return {"transformation": {("externalId" if camel_case else "external_id"): self.external_id}}
+        transformation = {("externalId" if camel_case else "external_id"): self.external_id}
```
Comment (haakonvt) [inline cognite/client/data_classes/workflows.py:235]:
```suggestion
        transformation = {"externalId" if camel_case else "external_id": self.external_id}
```

File: cognite/client/data_classes/workflows.py [hunk 3]
```diff
@@ -193,29 +194,52 @@ def dump(self, camel_case: bool = False) -> dict[str, Any]:
...
+
+        if self.concurrency_policy is not None:
+            transformation[
+                ("concurrencyPolicy" if camel_case else "concurrency_policy")
+            ] = self.concurrency_policy.value
```
Comment (haakonvt) [inline cognite/client/data_classes/workflows.py:240]: 120 on the dot!
```suggestion
            transformation["concurrencyPolicy" if camel_case else "concurrency_policy"] = self.concurrency_policy.value
```

File: cognite/client/data_classes/workflows.py [hunk 4]
```diff
@@ -193,29 +194,50 @@ def dump(self, camel_case: bool = False) -> dict[str, Any]:
...
     Args:
         external_id (str): The external ID of the transformation to be called.
-
+        concurrency_policy (TransformationConcurrencyPolicy | None): The concurrency policy of the transformation.
     """
 
     task_type = "transformation"
 
-    def __init__(self, external_id: str) -> None:
+    def __init__(self, external_id: str, concurrency_policy: TransformationConcurrencyPolicy | None = None) -> None:
```
Comment (doctrino) [inline cognite/client/data_classes/workflows.py:222]: Lets go for a Literal.

**Review by doctrino — APPROVED** [review]
Good Work

**Pull request merged**